src/components/TextInput.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Field, propTypes } from 'redux-form';
+import { TextField } from 'redux-form-material-ui';
+
+
+const TextInput = ({ name, type, placeholder, ...props }) => (
+  <Field
+    name={name}
+    component={TextField}
+    type={type}
```

Comment:
Ok. So what happens is that any prop that is not recognized by `Field` is passed to the component. In the examples it is `input` and the `type` argument for `<input>` is documented [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).

In our case the component is a `TextField` and the type attribute has no meaning there. So it can go.